pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+
+        method_name = '_fake_' + methodname.lower()
+
+        method_name = getattr(self, method_name)
+        result = method_name(namespace, **params)
+
+        if self.verbose:
+            print('mock result %s' % result)
+
+        return result
+
+    def _mock_methodcall(self, methodname, localobject, Params=None, **params):
+        # pylint: disable=invalid-name
+        """
+        Mock the execution of WBEMConnection _methodcall which is the common
+        calling interface for invokemethod
+        """
+        if self.verbose:
+            print('mock_imethodcall method=%s, namespace=%s, '
+                  'response_params_rqd=%s\nparams=%s'
+                  % (methodname, localobject, Params, params))
+
+        result = self._fake_invokemethod(methodname,
+                                         localobject,
+                                         Params=Params,
+                                         **params)
+
+        if self.verbose:
+            print('mock result %s' % result)
+
+    ################################################################
+    #
+    #   Methods to insert data into fake repository
+    #
+    ################################################################
+
+    @staticmethod
+    def display_objects(obj_type, objects_dict):
+        """
+        Display a set of objects of obj_type from the dictionary defined
+        by the parameter objects_dict. obj_type is a string that defines the
+        type of object (instance, class, qualifier declaration).
+        """
+        if not objects_dict:
+            return
+        for ns, objects in six.iteritems(objects_dict):
+            print('namespace %s: %s %s' % (ns, len(objects), obj_type))
+            for obj in six.itervalues(objects):
+                print('obj %s' % obj)
+                print(obj.tomof())
+
+    def display_repository(self):
+        """
+        Display everything in repository
+        """
+        print('===============Repository====================')
+        self.display_objects('qualifierdeclarations', self.qualifiers)
+        self.display_objects('classes', self.classes)
+
+        # instances are special because the inner struct is a list
+        for ns in six.iterkeys(self.instances):
+            print('instances namespace %s' % ns)
+            insts = self.instances[ns]
+            for inst in insts:
+                print('path=%s' % inst.path)
+                print(inst.tomof())
+        print('============End Repository=================')
+
+    def compile_mof(self, mof_file, namespace=None):
+        """
+        Compile mof from files
+        """
+        def moflog(msg):
+            """Display message to moflog2"""
+            print(msg, file=self.logfile)
+        if not namespace:
+            namespace = 'root/cimv2'
+        moflog_file = 'moflog_fake_WBEMConnection.txt'
+        self.logfile = open(moflog_file, 'w')
+        mofcomp = MOFCompiler(MOFWBEMConnection(),
```

Comment:
The MOF file is compiled without using the existing local repo as a basis. The logic is currently: compile standalone into a new repo, then merge new repo into local repo.

I would have expected that the local repo is used as a basis. 

I think the logic should be: copy local repo, compile MOF into the copy, copy back. That allows to leave the local repo unchanged in case of compile errors.

Reply:
DISCUSSION.  Since we now copy over and copy back, we are essentially recreating the full
set of objects each time.
DISCUSSION

Reply:
Now shown in list of discussion points, below.